packages/app/src/components/UserSettingsProvider.js

```diff
@@ -0,0 +1,49 @@
+import React, { useContext, useState, useEffect, createContext } from 'react'
+import PropTypes from 'prop-types'
+import { useDataEngine } from '@dhis2/app-runtime'
+
+export const userSettingsQuery = {
+    resource: 'userSettings',
+    params: {
+        key: ['keyDbLocale', 'keyUiLocale', 'keyAnalysisDisplayProperty'],
+    },
+}
+
+export const UserSettingsCtx = createContext({})
+
+const UserSettingsProvider = ({ children }) => {
+    const [settings, setSettings] = useState([])
+    const engine = useDataEngine()
+
+    useEffect(() => {
+        async function fetchData() {
+            const { userSettings } = await engine.query({
+                userSettings: userSettingsQuery,
+            })
+
+            setSettings({
+                ...userSettings,
+                displayProperty: userSettings.keyAnalysisDisplayProperty,
```

Comment:
Is this property used by the app?

Reply:
Looks like this and the `extractUserSettings` function need to be tweaked to not repeat each other.